### PR TITLE
fixing #309 by OpenSourcing buffer persistence

### PR DESCRIPTION
**Pull request by adamdubiel**
buffer persistence opensourced

### CODE REVIEW

File: integration/src/integration/java/pl/allegro/tech/hermes/integration/AbstractFrontendShutdownTest.java [hunk 1]
```diff
@@ -8,23 +8,24 @@
...
 
-    HermesPublisher publisher;
-    HermesServer hermesServer;
+    protected HermesPublisher publisher;
+    protected HermesServer hermesServer;
 
     private HermesFrontend hermesFrontend;
 
-    @BeforeClass
+//    @BeforeClass
```
Comment (druminski) [inline integration/src/integration/java/pl/allegro/tech/hermes/integration/AbstractFrontendShutdownTest.java]: commented @BeforeClass & @AfterClass

File: .travis.yml [hunk 2]
```diff
@@ -4,8 +4,8 @@ language: java
 
 install: ./gradlew assemble
 script:
-  - "./gradlew check"
-  - "./gradlew -Ptests.timeout.multiplier=3 integrationTest"
+#  - "./gradlew check"
```
Comment (druminski) [inline .travis.yml]: disabled unit tests

File: docs/docs/configuration/buffer-persistence.md [hunk 3]
```diff
@@ -23,11 +21,18 @@ There is additional protection against flooding subscribers with outdated events
...
--------------------------------------- | ------------------------------------------------------ | --------------
-messages.local.storage.max.age.hours   | ignore messages in buffer that are older than N hours  | 72
-messages.local.storage.directory       | location of memory mapped files                        | /tmp/<tmp dir>
-messages.loading.wait.for.topics.cache | how much seconds to wait on startup for topics to load | 10
+Option                                          | Description                                            | Default value
+----------------------------------------------- | ------------------------------------------------------ | --------------
+frontend.messages.local.storage.enabled         | enable persistent buffer                               | true
+frontend.messages.local.storage.max.age.hours   | ignore messages in buffer that are older than N hours  | 72
+frontend.messages.local.storage.directory       | location of memory mapped files                        | /tmp/<tmp dir>
+frontend.messages.loading.wait.for.topics.cache | how much seconds to wait on startup for topics to load | 10
```
Comment (dankraw) [inline docs/docs/configuration/buffer-persistence.md]: how much -> how many

File: hermes-frontend/build.gradle [hunk 4]
```diff
@@ -6,6 +6,9 @@ dependencies {
     compile group: 'io.undertow', name: 'undertow-servlet', version: '1.2.12.Final'
     compile group: 'org.apache.kafka', name: 'kafka-clients', version: '0.8.2.0'
     compile group: 'org.apache.commons', name: 'commons-lang3', version: '3.0'
+    compile group: 'net.openhft', name: 'chronicle-map', version: '2.1.15'
+    compile group: 'commons-io', name: 'commons-io', version: '2.4'
+    compile group: 'com.jayway.awaitility', name: 'awaitility', version: '1.6.1'
```
Comment (chemicL) [inline hermes-frontend/build.gradle]: testCompile?

File: hermes-common/src/main/java/pl/allegro/tech/hermes/common/config/Configs.java [hunk 5]
```diff
@@ -89,6 +90,11 @@
     FRONTEND_SSL_TRUSTSTORE_PASSWORD("frontend.ssl.truststore.password", "password"),
     FRONTEND_SSL_TRUSTSTORE_FORMAT("frontend.ssl.truststore.format", "JKS"),
 
+    MESSAGES_LOCAL_STORAGE_ENABLED("frontend.messages.local.storage.enabled", true),
```
Comment (dankraw) [inline hermes-common/src/main/java/pl/allegro/tech/hermes/common/config/Configs.java]: shouldn't it be disabled by default?

Comment (adamdubiel) [inline hermes-common/src/main/java/pl/allegro/tech/hermes/common/config/Configs.java]: okay, let's disable this

Comment (dankraw) [inline hermes-common/src/main/java/pl/allegro/tech/hermes/common/config/Configs.java]: I mean, as a new/incubating feature :P

**Comment by dankraw** [comment]
:+1:

**Comment by adamdubiel** [comment]
local tests pass (test + integration test)

**Comment by druminski** [comment]
:+1:

**Pull request merged**